solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -0,0 +1,1386 @@
+     * @dev syncs the stored reserve balance for a given reserve with the real reserve balance
+     *
+     * @param _reserveToken    address of the reserve token
+     */
+    function syncReserveBalance(IERC20Token _reserveToken) internal {
+        uint256 reserveId = __reserveIds[_reserveToken];
+        uint256 balance = _reserveToken == ETH_RESERVE_ADDRESS
+            ? address(this).balance
+            : _reserveToken.balanceOf(address(this));
+        setReserveBalance(reserveId, balance);
+    }
+
+    /**
+     * @dev syncs all stored reserve balances
+     */
+    function syncReserveBalances() internal {
+        IERC20Token _reserveToken0 = __reserveTokens[0];
+        IERC20Token _reserveToken1 = __reserveTokens[1];
+        uint256 balance0 = _reserveToken0 == ETH_RESERVE_ADDRESS
+            ? address(this).balance
+            : _reserveToken0.balanceOf(address(this));
+        uint256 balance1 = _reserveToken1 == ETH_RESERVE_ADDRESS
+            ? address(this).balance
+            : _reserveToken1.balanceOf(address(this));
+        setReserveBalances(1, 2, balance0, balance1);
+    }
+
+    /**
+     * @dev syncs all stored reserve balances, excluding a given amount of ether from the ether reserve balance (if relevant)
+     *
+     * @param _value   amount of ether to exclude from the ether reserve balance (if relevant)
+     */
+    function syncReserveBalances(uint256 _value) internal {
+        IERC20Token _reserveToken0 = __reserveTokens[0];
+        IERC20Token _reserveToken1 = __reserveTokens[1];
+        uint256 balance0 = _reserveToken0 == ETH_RESERVE_ADDRESS
+            ? address(this).balance - _value
+            : _reserveToken0.balanceOf(address(this));
+        uint256 balance1 = _reserveToken1 == ETH_RESERVE_ADDRESS
+            ? address(this).balance - _value
+            : _reserveToken1.balanceOf(address(this));
+        setReserveBalances(1, 2, balance0, balance1);
+    }
+
+    /**
+     * @dev helper, dispatches the Conversion event
+     *
+     * @param _sourceToken     source ERC20 token
+     * @param _targetToken     target ERC20 token
+     * @param _trader          address of the caller who executed the conversion
+     * @param _amount          amount purchased/sold (in the source token)
+     * @param _returnAmount    amount returned (in the target token)
+     */
+    function dispatchConversionEvent(
+        IERC20Token _sourceToken,
+        IERC20Token _targetToken,
+        address _trader,
+        uint256 _amount,
+        uint256 _returnAmount,
+        uint256 _feeAmount
+    ) internal {
+        // fee amount is converted to 255 bits -
+        // negative amount means the fee is taken from the source token, positive amount means its taken from the target token
+        // currently the fee is always taken from the target token
+        // since we convert it to a signed number, we first ensure that it's capped at 255 bits to prevent overflow
+        assert(_feeAmount < 2**255);
```

Comment:
This should be something like 2**127 now instead, right?

Reply:
Hmmmm, probably, yes. Not sure in what way this "thing" is still relevant to us.

Reply:
that check isn't relevant in this pool.

Reply:
Fixed.